onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -124,10 +129,142 @@ def _parse_line(line):
 
     return (scalefacs, zpoints)
 
+# Adding ability to record intermediate graph outputs
+selected_types = ['Conv', 'MatMul'] # node types to extend
+
+def augment_graph(model):
+    '''
+    Adds ReduceMin and ReduceMax nodes to all Conv and MatMul nodes in
+    model and ensures their outputs are stored as part of the graph output
+        parameter model: loaded FP32 ONNX model to quantize
+        return: augmented ONNX model
+    '''
+    added_nodes = []
+    added_outputs = []
+    for node in model.graph.node:
+        if node.op_type in selected_types:
+            input_name = node.name
+            # Adding ReduceMin nodes
+            reduce_min_name = input_name + '_ReduceMin'
+            reduce_min_node = onnx.helper.make_node('ReduceMin', [input_name + ':0'],
+                            [reduce_min_name + ':0'], reduce_min_name, keepdims=0)
+            added_nodes.append(reduce_min_node)
+            added_outputs.append(helper.make_tensor_value_info(reduce_min_node.name + ':0', TensorProto.FLOAT, ()))
+            # Adding ReduceMax nodes
+            reduce_max_name = input_name + '_ReduceMax'
+            reduce_max_node = onnx.helper.make_node('ReduceMax', [input_name + ':0'],
+                            [reduce_max_name + ':0'], reduce_max_name, keepdims=0)
+            added_nodes.append(reduce_max_node)
+            added_outputs.append(helper.make_tensor_value_info(reduce_max_node.name + ':0', TensorProto.FLOAT, ()))
+    model.graph.node.extend(added_nodes)
+    model.graph.output.extend(added_outputs)
+    return model
+
+# Reading user input (images)
+def load_and_resize_image(image_filepath, height, width):
+    '''
+    Resizes image to  to NCHW format
+        parameter image_filepath: path to image files
+        parameter height: image height in pixels
+        parameter width: image width in pixels
+        return: matrix characterizing image
+    '''
+    pillow_img = Image.open(image_filepath).resize((width, height))
+    input_data = np.float32(pillow_img)/127.5 - 1.0 # normalization
+    input_data -= np.mean(input_data)
```

Comment:
did you check what is the effect of this on min max?
Also can you call out both these steps as normalization...

Reply:
Done! (seems to decrease the magnitudes of the min and max values further)